README.md

```diff
@@ -99,77 +103,110 @@ docker run \
     ...
 ```
 
+## 3.3. Specific Configuration Options
+
+| Name                              | Type            | Default Value | Description                                      |
+|:----------------------------------|:----------------|:--------------|:-------------------------------------------------|
+| storage-driver-create-key-enabled | boolean         | false         | Specifies if Mongoose should generate its own routing key during the events creation
+| storage-driver-create-key-count   | integer         | 0             | Specifies a max count of unique routing keys to use during the events creation (may be considered as a routing key period). 0 value means to use unique routing key for each new event
+| storage-driver-read-timeoutMillis | integer         | 100           | The event read timeout in milliseconds
+| storage-net-node-addrs            | list of strings | 127.0.0.1     | The list of the Pravega storage nodes to use for the load
+| storage-net-node-port             | integer         | 9020          | The default port of the Pravega storage nodes, should be explicitly set to 9090 (the value used by Pravega by default)
+
 # 4. Design
 
-Mongoose and Pravega are using different concepts. So it's necessary to determine how
+Mongoose and Pravega are using quite different concepts. So it's necessary to determine how
 [Pravega-specific terms](http://pravega.io/docs/latest/terminology/) are mapped to the
 [Mongoose abstractions]((https://github.com/emc-mongoose/mongoose/tree/master/doc/design/architecture#1-basic-terms)).
 
 | Pravega | Mongoose |
 |---------|----------|
 | [Stream](http://pravega.io/docs/latest/pravega-concepts/#streams) | *Path Item* |
-| Scope | *Storage Path* part (*stream name* is the 2nd part) |
+| Scope | Storage Namespace
 | [Event](http://pravega.io/docs/latest/pravega-concepts/#events) | *Data Item* |
 | Stream Segment | N/A |
 
-**Note**:
-> The Pravega storage driver should extend the NIO storage driver base.
-
-## 4.1. Data Item Operations
+## 4.1. Event Operations
 
 Mongoose should perform the load operations on the *events* when the configuration option `item-type` is set to `data`.
 
 ### 4.1.1. Create
 
-1. Check the corresponding scope if it exists and create it if not.
-2. Check the corresponding stream if it exists and create it if not.
-1. Get an `EventStreamWriter<ByteBuffer>` instance using the *item id* as a *stream name*.
-2. Submit the next `ByteBuffer` event writing using a `writeEvent` method with a routing key either without it. Note
-that it returns the `CompletableFuture` which should be handled properly.
+Steps:
+1. Get the endpoint URI from the cache.
+2. Check if the corresponding `StreamManager` exists using the cache, create a new one if it doesn't.
+3. Check if the destination scope exists using the cache, create a new one if it doesn't.
```

Comment:
I thought we agreed that scope is a configuration parameter. How can it happen that it doesn't exist?
or am i missing something?

Reply:
An user able to configure the name of the scope to use. It may not exist on the storage side.

Reply:
Got you